data-processing/entities/definitions/commands/detect_definitions.py

```diff
@@ -377,6 +381,24 @@ class SymbolNickname(NamedTuple):
     nickname_indexes: List[int]
     nickname_ranges: List[CharacterRange]
 
+def get_filtered_symbol_nickname_pairs(
+    pairs: List[SymbolNickname],
+    text: str
+) -> List[SymbolNickname]:
+    # Filter symbol-nickname pairs based on the following conditions.
+    # 1. The nickname of a symbol is another symbol.
+    filtered_pairs: List[SymbolNickname] = []
+    for sn in pairs:
+        nickname_start = min(
+            [nickname_range.start for nickname_range in sn.nickname_ranges]
+        )
+        nickname_end = (
+            max([nickname_range.end for nickname_range in sn.nickname_ranges]) + 1
+        )
```

Comment:
nit pick: Converting text ranges into text like something the nickname filtering function shouldn't have to take care of. Might it make sense to extract the text for a nickname when the SymbolNickname is getting created in `get_symbol_nickname_pairs`, and just inspect the text here?

Reply:
Yeah, this is true. But I was wondering if we want to add other filters as well that might not just depend on the nickname text. But we can maybe have a boolean function.

Reply:
Fair enough. Do you think they would depend on the character ranges? Maybe it's worth us still attaching the text to the SymbolNickname to support easier checks in this filter function, while keeping the range properties on the nickname too in case we want to add that in the future..